pkg/storage/cacher.go

```diff
@@ -402,10 +403,20 @@ func filterFunction(key string, keyFunc func(runtime.Object) (string, error), fi
 			glog.Errorf("invalid object for filter: %v", obj)
 			return false
 		}
-		if !strings.HasPrefix(objKey, key) {
-			return false
+
+		segments := strings.Split(objKey, "/")
+		currPath := ""
+
+		match := false
+		for _, segment := range segments {
+			currPath = path.Join(currPath, segment)
+			if currPath == key {
```

Comment:
It's fine here because the results are already filtered out by underlying storage impl.?